### PR TITLE
DISCOUNT add Discount Feature

### PR DESCRIPTION
**Pull request by Mo7ammadAbuSafat**

### CODE REVIEW

**Comment by MazenAmria** [comment]
avoid using force-push, it's a bad idea when you're working with a team
and it's ok if the commit needs some changes, add another one.

git is a source version control software, so it's supposed to preserve history of the code, so it's ok to have multiple small changes in different commits. That's why a PR exists, all the related changes (commits) are grouped in one PR. Some teams might go further and uses what's is called git squash, where they reduce all the changes of the PR into single commit on the master instead of merging the commits as they are.

File: Product.cs [hunk 1]
```diff
@@ -20,41 +22,40 @@ public double Price
...
-            {
-                if (value < 0)
-                {
-                    throw new ArgumentException("Tax percentage cannot be a negative value.");
-                }
-
-                _taxPercentage = value;
-            }
-        }
+        public static PriceCalculator priceCalculator = new PriceCalculator();
```
Comment (MazenAmria) [inline Product.cs]: Imagine I have 4 products, and I wanted to change the tax percentage, or the discount percentage.
I need to iterate over all products and change it for them. Is it practical?

File: PriceCalculator.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        public double CalculateTax(double price)
+        {
+            return Math.Round(TaxPercentage * price, 2);
+        }
+
+        public double CalculateDiscount(double price)
+        {
+            return Math.Round(DiscountPercentage * price, 2);
+
+        }
```
Comment (MazenAmria) [inline PriceCalculator.cs:58]: add another method to calculate the total price (which uses both methods to get it done).

**Review by MazenAmria — CHANGES_REQUESTED** [review]

File: Product.cs (hunk 1 above)
Comment (Mo7ammadAbuSafat) [reply Product.cs]: in the requirements the tax an discount is equal for all products

File: PriceCalculator.cs (hunk 2 above)
Comment (Mo7ammadAbuSafat) [reply PriceCalculator.cs:58]: ok I will

File: Product.cs (hunk 1 above)
Comment (MazenAmria) [reply Product.cs]: @Mo7ammadAbuSafat Ok so I see you've added the price calculator as a static field.

**Review by MazenAmria — APPROVED** [review]
just add a Math.Round for the tax and discount setters, other than that it's all good.

**Pull request merged**